tests/test_two_sided_market.py

```diff
@@ -495,3 +506,30 @@ def test_match_recommendations_multiple_bids_offers(self, market):
         assert market.accumulated_trade_price == 2
         assert market.traded_energy["Seller"] == 2
         assert market.traded_energy["Buyer"] == -2
+
+    def test_double_sided_pay_as_clear_market_area_retain_state(self, pac_area):
+
+        offers = [
+            Offer("id1", now(), 1.1, 1, "other").serializable_dict(),
+            Offer("id2", now(), 2.2, 1, "other").serializable_dict(),
+            Offer("id3", now(), 3.3, 1, "other").serializable_dict()]
+        bids = [
+            Bid("bid_id1", now(), 3.3, 1, "B", "S").serializable_dict(),
+            Bid("bid_id2", now(), 2.2, 1, "B", "S").serializable_dict(),
+            Bid("bid_id3", now(), 1.1, 1, "B", "S").serializable_dict()]
+        current_time = pac_area.now
+
+        data = {str(uuid4()): {"bids": bids, "offers": offers, "current_time": current_time}}
+        bid_offer_matcher.matcher._get_matches_recommendations(data)
+
+        pac_area.copy_clearing_state_to_area_state()
+
+        assert (pac_area.state.cumulative_offers[current_time] ==
+                bid_offer_matcher.matcher.match_algorithm.state.cumulative_offers)
+        assert (pac_area.state.cumulative_bids[current_time] ==
+                bid_offer_matcher.matcher.match_algorithm.state.cumulative_bids)
+
+    @classmethod
+    def teardown_class(cls):
```

Comment:
This is still not functional
You can edit your pac_area like the following
```python
@pytest.fixture
def pac_area():
    ConstSettings.IAASettings.MARKET_TYPE = 2
    ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE = BidOfferMatchAlgoEnum.PAY_AS_CLEAR.value
    bid_offer_matcher.activate()
    yield Area(name="fake_area")
    ConstSettings.IAASettings.MARKET_TYPE = 1
    ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE = BidOfferMatchAlgoEnum.PAY_AS_BID.value
```

Reply:
Thats what the link your shared earlier proposed it

Reply:
Not really, I proposed to use `pytest fixtures` not typical class controllers

Reply:
Okay, using `pytest fixtures` now https://github.com/gridsingularity/d3a/pull/1195/commits/f573a8f642ea11aee1d0e4cca2af393bd0b8cec7